### DISCUSSION TITLE
Does CKAN supports multiple applications for authentication?

### DISCUSSION THREAD

**Discussion by ArqamFarooqui110719** [ckan/ckan#6981] [Q&A] [answered]
Hi all, 
I'm using dockerized CKAN(2.7.2).

Does CKAN supports multiple applications (for e.g. application of primary tenant and application from secondary tenant at a time) for authentication? I'm using WSO2 for the authentication. 

![ckan-oauth-configuration](https://user-images.githubusercontent.com/55977298/180358343-09b63b1f-9e19-4b1c-b5a9-b92cb5585c93.PNG)

For e.g.
-> I've created a user `test1` as a primary tenant user and `test2` as a secondary tenant user in wso2. And I've used the `client_id` and `client_secret` of primary tenant user (test1) in ckan.ini file (screentshot is attached above).

In the above case when we are login the CKAN with `test1` (primary user), then we are successfully able to login but when we are login the CKAN with `test2` (secondary user). then we are getting error i.e. `Application you are trying to access does not allow users from your organization`.

How can we login from both users (multiple users from different tenants)? or how CKAN supports multiple application at a time?

**Comment by ArqamFarooqui110719** [comment]
Hi @amercader 
Can you please suggest, "Does CKAN supports multiple applications at same time created on wso2?"

**Reply by amercader** [reply]
This is not core CKAN functionality, you'll need to check with the maintainers of ckanext-oauth2 (or whatever extension you are using for this).
Also please upgrade your CKAN version, at least to the latest patch release, ideally to CKAN 2.9

**Comment by ArqamFarooqui110719** [comment, marked as answer]
Issue was with wso2, now resolved:)
We can login from both primary and secondary users created on WSO2. I was facing issue bcoz users were not in sync.
